### PR TITLE
Feat/yoink gh pages workflow

### LINKED ISSUE #232: It's GitHub pages time.
Using a workflow and rustdoc, write some documentation that doesn't reside in the README.

Yoink [rust-analyzer](https://github.com/rust-lang/rust-analyzer)'s GitHub pages workflow.

### PR DESCRIPTION
**Pull request by Nadhum**
Yoinked!!

## Description
Setting up (yoinking) a rust doc workflow and publishing it to gh pages.

Note: need to set up github pages with the `main` as the branch and `docs` as the root.

Closes #232

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -109,6 +109,10 @@
 
 </details>
 
+## Links
+
+- rustdoc docs: https://xithrius.github.io/twitch-tui/twt/
```
[the remark this answers is not in the document]
Comment (Xithrius) [reply README.md:114]: Nevermind.

File: .github/workflows/rustdoc.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      run: rustup update --no-self-update stable
+
+    - name: Build Documentation
+      run: cargo doc --all --no-deps
+
+    - name: Deploy Docs
+      uses: peaceiris/actions-gh-pages@v3
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        publish_branch: gh-pages
```
Comment (Xithrius) [inline .github/workflows/rustdoc.yml:32]:
```suggestion
        publish_branch: main
```

Don't change this yet, but I need to think about how this should actually be approached.

Perhaps a `docs` folder will be created in the main branch.

Comment (Nadhum) [reply .github/workflows/rustdoc.yml:37]: Yup a docs folder will be needed on the main branch

File: README.md (hunk 1 above)
Comment (Xithrius) [inline README.md:114]: I don't believe we'll need the binary name there.

```suggestion
- Documentation: https://xithrius.github.io/twitch-tui/
```

Comment (Nadhum) [reply README.md:114]: @Xithrius index.html is generated under target/doc/twt so I think we need it, based on my testing at least

File: .github/workflows/rustdoc.yml (hunk 2 above)
Comment (Nadhum) [reply .github/workflows/rustdoc.yml:37]: @Xithrius should we change this to be the main branch? or keep it in a separate branch?
A separate branch seems cleaner to me tbh, what do you think?

Comment (Xithrius) [reply .github/workflows/rustdoc.yml:37]: I like having everything in one place, so let's do main. If it's decided later that this isn't the way to do this project, then the docs can be moved to a new branch.

File: README.md (hunk 1 above)
Comment (Xithrius) [reply README.md]: hmmmmmmmmm

Alright. Let's at least change this to the following:

```md
- Documentation: https://xithrius.github.io/twitch-tui/twt
```

File: .github/workflows/rustdoc.yml [hunk 3]
```diff
@@ -30,10 +30,13 @@ jobs:
     - name: Build Documentation
       run: cargo doc --all --no-deps
 
+    - name: Add index redirection
+      run: echo "<meta http-equiv=\"refresh\" content=\"0; url=twt\">" > ./target/doc/index.html
+
```
Comment (Nadhum) [inline .github/workflows/rustdoc.yml:35]: Adding this index.html under docs so we can link to the docs by `https://xithrius.github.io/twitch-tui/`

Comment (Xithrius) [reply .github/workflows/rustdoc.yml:35]: Very cool.

File: .github/workflows/rustdoc.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      run: cargo doc --all --no-deps
+
+    - name: Add index redirection
+      run: echo "<meta http-equiv=\"refresh\" content=\"0; url=twt\">" > ./target/doc/index.html
+
+    - name: Deploy Docs
+      uses: peaceiris/actions-gh-pages@v3
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        publish_branch: main
```
Comment (Xithrius) [inline .github/workflows/rustdoc.yml:40]: Oh dear. After generating the docs locally I realize that the sheer amount of stuff is too much to be on the main branch.

Sorry for bikeshedding, I'll merge this PR after this change. Thanks for contributing!

```suggestion
        publish_branch: gh-pages
```

**Comment by Xithrius** [comment]
Also, if you want to do another PR, feel free to create one which implements https://github.com/rust-lang/mdBook, [shown here](https://github.com/peaceiris/actions-gh-pages#%EF%B8%8F-mdbook-rust).

**Comment by Nadhum** [comment]
> If you want to do another PR, feel free to create one which implements https://github.com/rust-lang/mdBook with a workflow, [shown here](https://github.com/peaceiris/actions-gh-pages#%EF%B8%8F-mdbook-rust).
> 
> Or if you'd like to do it now and this PR, feel free to do that as well.

We want both rustdoc + mdbook? or discard this rustdoc?

**Comment by Xithrius** [comment]
Since I am a dingus, I realized that rustdoc wouldn't be that helpful in developing this application.

Also in the start of this I did think that rustdoc *was* mdbook. I didn't know either existed until a short while ago, somehow.

**Comment by Nadhum** [comment]
> Since I am a dingus, I realized that rustdoc wouldn't be that helpful in developing this application.
> 
> Also in the start of this I did think that rustdoc _was_ mdbook. I didn't know either existed until a short while ago, somehow.
> 
> So we should probably discard rustdoc and go for mdbook, since that's the type of documentation I was going for in the first place.
> 
> I'm sorry for going back and forth with this, it's kind of ridiculous.

No worries, done, also put some place holder docs for the moment.

**Review by Xithrius — APPROVED** [review]
Very good. I apologize once again for the bikeshedding.

Thank you for contributing to this PR. Have a good one!

**Comment by Xithrius** [comment]
And just like that, [it works!](https://xithrius.github.io/twitch-tui/)

**Comment by Nadhum** [comment]
Hahah such magic!

**Pull request merged**